pkg/kademlia/kademlia_test.go

```diff
@@ -301,7 +317,7 @@ func TestGetNodes(t *testing.T) {
 			}
 		})
 	}
-
+	defer cleanup(ctx, t, k, dir)
```

Comment:
This defer should be near the construction of these things rather than the last thing. Also, defer as the last statement rarely makes sense.